src/viewers/global-selection-hook.js

```diff
@@ -0,0 +1,30 @@
+import { useState } from 'react'; // eslint-disable-line import/no-unresolved
+import { GLOBAL_SLIDER_DIMENSION_FIELDS } from '../constants';
+
+export default function useGlobalSelection(loaderSelection) {
+  // viewportSelection is the last selection that had an onViewportLoad callback.
+  const [viewportSelection, setViewportSelection] = useState(loaderSelection);
+  let onViewportLoad;
+  let newLoaderSelection = null;
+  let oldLoaderSelection = loaderSelection;
+  if (
+    loaderSelection?.length &&
+    viewportSelection?.length &&
+    GLOBAL_SLIDER_DIMENSION_FIELDS.some(
```

Comment:
In the spirit of having `transitionFields` be a react prop, we would make this customizable as well.